src/borg/helpers/fs.py

```diff
@@ -19,6 +21,33 @@
 py_37_plus = sys.version_info >= (3, 7)
 
 
+def ensure_dir(path, name=None, mode=None, reraise=False):
```

Comment:
maybe rename the `reraise` to `pretty_deadly=True`.

the main difference of `Error` exception is that it does not include a traceback (thus it is "pretty") and also it usually terminates the application (thus it is "deadly").

Then, in the exception handlers:
```
if pretty_deadly:
    raise Error(...)
else:
    raise
```